### PR TITLE
Typeahead refactor -- remove (old) search

### PR DESCRIPTION
**Pull request by raymondng**
Clean-up#1. More refactoring to come.

RIP search (older method)

@ymatsuda @eishay

### CODE REVIEW

**Comment by raymondng** [comment]
reviewed with @ymatsuda

File: kifi-backend/modules/common/app/com/keepit/typeahead/Typeahead.scala [hunk 1]
```diff
@@ -49,26 +49,6 @@ trait Typeahead[E, I] extends Logging {
 
   protected def extractName(info: I): String
 
-  def search(userId: Id[User], query: String)(implicit ord: Ordering[TypeaheadHit[I]]): Option[Seq[I]] = timing(s"search($userId,$query)") {
```
Comment (eishay) [inline kifi-backend/modules/common/app/com/keepit/typeahead/Typeahead.scala:52]: dead code is good code :-)

Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/typeahead/Typeahead.scala:52]: lol kifi zen :)

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminUserController.scala [hunk 2]
```diff
@@ -690,20 +690,22 @@ class AdminUserController @Inject() (
   }
 
   // ad hoc testing only during dev phase
-  private def prefixSocialSearchDirect(userId:Id[User], query:String):Option[Seq[SocialUserBasicInfo]] = {
+  private def prefixSocialSearchDirect(userId:Id[User], query:String): Future[Option[Seq[SocialUserBasicInfo]]] = {
```
Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminUserController.scala:693]: i like methods that return a future.
do we need to have an `Option[Seq[*]]`? Isn't an empty `Seq` is the same as None in this case?

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminUserController.scala [hunk 3]
```diff
@@ -690,20 +690,22 @@ class AdminUserController @Inject() (
...
-    val resOpt = prefixSocialSearchDirect(userId, query)
-    resOpt match {
-      case None =>
-        Ok(s"No social match found for $query")
-      case Some(res) =>
-        Ok(res.map{ info => s"SocialUser: id=${info.id} name=${info.fullName} network=${info.networkType} <br/>" }.mkString(""))
+  def prefixSocialSearch(userId:Id[User], query:String) = AdminHtmlAction.authenticatedAsync { request =>
+    prefixSocialSearchDirect(userId, query) map { resOpt =>
+      resOpt match {
+        case None =>
```
Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminUserController.scala:704]: i.e. `case Seq()` in this case?

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminUserController.scala (hunk 2 above)
Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminUserController.scala:693]: yup that's internal details that's not going to be useful to clients. 
as mentioned, more to come ... just getting started. note that this one is not exposed as an api -- i may kill some admin endpoints in this process but i'd also like to keep things manageable.

Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminUserController.scala:693]: details: none = no filter; empty seq = no results for query. it's not random but it's low-level.

**Pull request merged**